Library/Homebrew/test/api/internal_tap_json/formula_spec.rb

```diff
@@ -0,0 +1,128 @@
+# frozen_string_literal: true
+
+describe "Internal Tap JSON -- Formula" do
+  let(:internal_tap_json) { File.read(TEST_FIXTURE_DIR/"internal_tap_json/homebrew-core.json").chomp }
+  let(:tap_git_head) { "9977471165641744a829d3e494fa563407503297" }
+
+  context "when generating JSON", :needs_macos do
+    before do
+      cp_r(TEST_FIXTURE_DIR/"internal_tap_json/homebrew-core", Tap::TAP_DIRECTORY/"homebrew")
+
+      # NOTE: Symlinks can't be copied recursively so we create them manually here.
+      (Tap::TAP_DIRECTORY/"homebrew/homebrew-core").tap do |core_tap|
+        mkdir(core_tap/"Aliases")
+        ln_s(core_tap/"Formula/f/fennel.rb", core_tap/"Aliases/fennel-lang")
+        ln_s(core_tap/"Formula/p/ponyc.rb", core_tap/"Aliases/ponyc-lang")
+      end
+    end
+
+    it "creates the expected hash" do
+      api_hash = CoreTap.instance.to_api_hash
```

Comment:
It looks like this line is picking up some unrelated formulae at test time. I guess that means that the test tap is not isolated from test to test so I might have to change my approach here.

My first approach was trying to mock the core tap directory so that I wouldn't have to move and write the files in the first place but I couldn't get the paths to work so I fell back on this approach.

Reply:
```
Failures:

  1) Internal Tap JSON -- Formula when generating JSON creates the expected hash
     Failure/Error: api_hash = CoreTap.instance.to_api_hash

     FormulaUnavailableError:
       No available formula with the name "testball".
     # ./formulary.rb:669:in `get_formula'
     # ./formulary.rb:758:in `factory'
     # ./tap.rb:1182:in `block in to_api_hash'
     # ./tap.rb:1181:in `to_h'
     # ./tap.rb:1181:in `to_api_hash'
     # ./test/api/internal_tap_json/formula_spec.rb:20:in `block (3 levels) in <top (required)>'
     # ./test/support/helper/spec/shared_context/homebrew_cask.rb:64:in `block (2 levels) in <top (required)>'
```